python/cudf/cudf/tests/dataset_generator.py

```diff
@@ -131,30 +131,47 @@ def _generate_column(column_params, num_rows):
         else:
             arrow_type = None
 
-        vals = pa.array(
-            column_params.generator,
-            size=column_params.cardinality,
-            safe=False,
-            type=arrow_type,
-        )
-        # Generate data for current column
-        return pa.array(
-            np.random.choice(vals, size=num_rows),
-            mask=np.random.choice(
-                [True, False],
+        if isinstance(arrow_type, pa.lib.Decimal128Type):
+            vals = pa.array(
+                np.random.choice(column_params.generator, size=num_rows),
+                mask=np.random.choice(
+                    [True, False],
+                    size=num_rows,
+                    p=[
+                        column_params.null_frequency,
+                        1 - column_params.null_frequency,
+                    ],
+                )
+                if column_params.null_frequency > 0.0
+                else None,
                 size=num_rows,
-                p=[
-                    column_params.null_frequency,
-                    1 - column_params.null_frequency,
-                ],
+                safe=False,
+            )
+            vals = vals.cast(arrow_type, safe=False)
+            return vals
+        else:
+            vals = pa.array(
+                column_params.generator,
+                size=column_params.cardinality,
+                safe=False,
+                type=arrow_type,
+            )
+            return pa.array(
+                np.random.choice(vals, size=num_rows),
+                mask=np.random.choice(
+                    [True, False],
+                    size=num_rows,
+                    p=[
+                        column_params.null_frequency,
+                        1 - column_params.null_frequency,
+                    ],
+                )
+                if column_params.null_frequency > 0.0
+                else None,
+                size=num_rows,
+                safe=False,
+                type=arrow_type,
```

Comment:
Can we avoid some code duplication here with just using `type=arrow_type if not isinstance(arrow_type, pa.lib.Decimal128Type) else None`, and then casting if necessary?

Reply:
Changed it 👍